modules/pam_unix/pam_unix_passwd.c

```diff
@@ -346,11 +346,11 @@ static int check_old_password(const char *forwho, const char *newpass)
 	size_t n = 0;
 	size_t len = strlen(forwho);
 
-	opwfile = fopen(OLD_PASSWORDS_FILE, "r");
+	opwfile = fopen(OLD_PASSWORDS_FILE, "re");
 	if (opwfile == NULL)
 		return PAM_ABORT;
 
-	while (getline(&buf, &n, opwfile) != -1) {
+	for (; getline(&buf, &n, opwfile) != -1; pam_overwrite_n(buf, n)) {
```

Comment:
We might call `pam_overwrite_n` here, but it's not guaranteed that `realloc` within `getline` wipes the previous memory area. Not saying that this change should be removed, but it should be considered if reallocation and memory analysis is an issue here.

As a side note: Same applies to https://github.com/linux-pam/linux-pam/pull/696

Reply:
Since `pam_overwrite_n()` gets called in the iteration-expression `realloc(3)` in the condition-expression always operates on cleaned memory.

Reply:
It's uncertain how often getline has called realloc internally, e.g. on initial long line. It could be like 32->64->128 for a 100 character long line. It really depends on the getline implementation.

Also, it would depend on the FILE handling of the library if the internal FILE buffer is cleaned, so while this is probably the best attempt that can be done, it's probably not sufficient.